neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+                    segment_name = self._file.units.segment[id]
+                    block_name = self._file.units.block[id]
+                except AttributeError:
+                    # For NWB files created with other applications, we put everything in a single
+                    # segment in a single block
+                    segment_name = "default"
+                    block_name = "default"
+                segment = self._get_segment(block_name, segment_name)
+                spiketrain = SpikeTrainProxy(self._file.units, id)
+                if not lazy:
+                    spiketrain = spiketrain.load()
+                segment.spiketrains.append(spiketrain)
+                spiketrain.segment = segment
+
+    def _read_acquisition_group(self, lazy):
+        self._read_timeseries_group("acquisition", lazy)
+
+    def _read_stimulus_group(self, lazy):
+        self._read_timeseries_group("stimulus", lazy)
+
+    def write_all_blocks(self, blocks, **kwargs):
+        """
+        Write list of blocks to the file
+        """
+        # todo: allow metadata in NWBFile constructor to be taken from kwargs
+        annotations = defaultdict(set)
+        for annotation_name in GLOBAL_ANNOTATIONS:
+            if annotation_name in kwargs:
+                annotations[annotation_name] = kwargs[annotation_name]
+            else:
+                for block in blocks:
+                    if annotation_name in block.annotations:
+                        try:
+                            annotations[annotation_name].add(block.annotations[annotation_name])
+                        except TypeError:
+                            if annotation_name in POSSIBLE_JSON_FIELDS:
+                                encoded = json.dumps(block.annotations[annotation_name])
+                                annotations[annotation_name].add(encoded)
+                            else:
+                                raise
+                if annotation_name in annotations:
+                    if len(annotations[annotation_name]) > 1:
+                        raise NotImplementedError(
+                            "We don't yet support multiple values for {}".format(annotation_name))
+                    # take single value from set
+                    annotations[annotation_name], = annotations[annotation_name]
+        if "identifier" not in annotations:
+            annotations["identifier"] = self.filename
+        if "session_description" not in annotations:
+            annotations["session_description"] = blocks[0].description or self.filename
+            # todo: concatenate descriptions of multiple blocks if different
+        if "session_start_time" not in annotations:
+            raise Exception("Writing to NWB requires an annotation 'session_start_time'")
+        # todo: handle subject
+        # todo: store additional Neo annotations somewhere in NWB file
```

Comment:
are these todos for this or the next PR? If they are going to be postponed, would it make sense to collect all of those options for improvements in a single issue?